bluemira/builders/tf_coils.py

```diff
@@ -641,44 +508,53 @@ def __init__(
             params.z_0.value,
             params.B_0.value,
         )
-        ripple_constraint = ripple_selector.make_ripple_constraint(
+        self._ripple_constraint = ripple_selector.make_ripple_constraint(
             parameterisation, self.solver, params.TF_ripple_limit.value, rip_con_tol
         )
         self.ripple_selector = ripple_selector
 
-        constraints = [ripple_constraint]
-
-        if keep_out_zone is not None:
-            koz_points = self._make_koz_points(keep_out_zone)
-            koz_constraint = OptimisationConstraint(
-                constrain_koz,
-                f_constraint_args={
-                    "parameterisation": parameterisation,
-                    "n_shape_discr": n_koz_points,
-                    "koz_points": koz_points,
-                },
-                tolerance=koz_con_tol * np.ones(n_koz_points),
-            )
-            constraints.append(koz_constraint)
-
-        super().__init__(parameterisation, optimiser, objective, constraints)
-
     def _make_koz_points(self, keep_out_zone: BluemiraWire) -> Coordinates:
         """
         Make a set of points at which to evaluate the KOZ constraint
         """
+        # TODO: I guess I'm gonna lose this fine control?
         return keep_out_zone.discretize(byedges=True, dl=keep_out_zone.length / 200).xz
 
+    def objective(self, parameterisation: GeometryParameterisation) -> float:
+        """
+        Objective function (minimise length)
+        """
+        return parameterisation.create_shape().length
+
+    def keep_out_zones(self):
+        """
+        Keep out zone
+        """
+        return self._keep_out_zone
+
+    def ineq_constraints(self):
+        """
+        Inequality constraints
+        """
+        return [self._ripple_constraint]
+
     def optimise(self, x0: Optional[np.ndarray] = None) -> GeometryParameterisation:
         """
         Solve the GeometryOptimisationProblem.
         """
-        parameterisation = super().optimise(x0=x0)
-        self.solver.update_cage(parameterisation.create_shape())
+        result = super().optimise(
+            self.parameterisation,
+            algorithm=self.algorithm,
+            opt_conditions=self.opt_conditions,
+            opt_parameters=self.opt_parameters,
+        )
+        self.parameterisation.variables.set_values_from_norm(result.x)
```

Comment:
Just so you're aware, there's some debate over whether we want to optimise geometries in-place, or return an optimised copy*. At the moment, we unfortunately can't copy an `OptVariables` instance (therefore we can't copy a geometry parameterisation, but Ollie is doing some work toward fixing this), so we're always editing in place. Which means, for the time being, I don't think this call is doing anything. 

I've also been thinking about whether having the `x` in the geometry optimisation result makes sense. Separated from their bounds, the normalised values don't really mean anything. The key output of the geometry optimisation was really intended to be the `.geom` attribute of the result.

With the above considered, I think the most future-proof way of doing this would be:
```suggestion
        self.parameterisation = super().optimise(
            self.parameterisation,
            algorithm=self.algorithm,
            opt_conditions=self.opt_conditions,
            opt_parameters=self.opt_parameters,
        ).geom
```
\
\
\* My preference is to take a copy. When we pass `x0` into a normal optimisation, we don't expect it to be updated in place, so why should we expect the geometry to be update in place?